agent/consul/leader_connect_ca.go

```diff
@@ -1004,6 +1007,7 @@ func (c *CAManager) UpdateConfiguration(args *structs.CARequest) (reterr error)
 	args.Op = structs.CAOpSetRootsAndConfig
 	args.Index = idx
 	args.Config.ModifyIndex = confIdx
+	args.Cas = confIdx
```

Comment:
Same here
I set `cas` to ModifyIndex in here because we are reading modifying and writing the config in one place, I'm not sure about this may be I need to set it to 0 to avoid the check.